eclipse-collections-code-generator/src/main/resources/impl/list/mutable/unmodifiablePrimitiveList.stg

```diff
@@ -237,6 +241,22 @@ public final class Unmodifiable<name>List
     {
         throw new UnsupportedOperationException("subList not yet implemented!");
     }
+
+    /**
+     * @since 9.1.
+     */
+    public MutableList\<<name><name>Pair> zip<name>(<name>List list)
+    {
+        return this.getMutable<name>List().zip<name>(list);
```

Comment:
Should we return an UnmodifiableList?

Reply:
I believe this is consistent with other methods.  For example, select:

```
    @Override
    public Mutable<name>List select(<name>Predicate predicate)
    {
        return this.getMutable<name>List().select(predicate);
    }
```

Reply:
ok makes sense.